testbed/tests/log_test.go

```diff
@@ -233,3 +241,116 @@ func TestLogOtlpSendingQueue(t *testing.T) {
 	})
 
 }
+
+func TestLogLargeFiles(t *testing.T) {
+	tests := []struct {
+		name         string
+		sender       testbed.DataSender
+		receiver     testbed.DataReceiver
+		loadOptions  testbed.LoadOptions
+		sleepSeconds int
+	}{
+		{
+			name:     "filelog-largefiles-2Gb-lifetime",
+			sender:   datasenders.NewFileLogWriter(),
+			receiver: testbed.NewOTLPDataReceiver(testutil.GetAvailablePort(t)),
+			loadOptions: testbed.LoadOptions{
+				DataItemsPerSecond: 200000,
+				ItemsPerBatch:      1,
+				Parallel:           100,
+			},
+			sleepSeconds: 100,
+		},
+		{
+			name:     "filelog-largefiles-6GB-lifetime",
+			sender:   datasenders.NewFileLogWriter(),
+			receiver: testbed.NewOTLPDataReceiver(testutil.GetAvailablePort(t)),
+			loadOptions: testbed.LoadOptions{
+				DataItemsPerSecond: 330000,
+				ItemsPerBatch:      10,
+				Parallel:           10,
+			},
+			sleepSeconds: 200,
+		},
+		{
+			name:     "filelog-largefiles-50MB/sec",
+			sender:   datasenders.NewFileLogWriter(),
+			receiver: testbed.NewOTLPDataReceiver(testutil.GetAvailablePort(t)),
+			loadOptions: testbed.LoadOptions{
+				DataItemsPerSecond: 400000,
+				ItemsPerBatch:      100,
+				Parallel:           1,
+			},
+			sleepSeconds: 100,
+		},
+	}
+	processors := map[string]string{
+		"batch": `
+  batch:
+`}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ScenarioLong(
+				t,
+				test.sender,
+				test.receiver,
+				test.loadOptions,
+				performanceResultsSummary,
+				test.sleepSeconds,
+				processors,
+			)
+		})
+	}
+}
+
+func TestLargeFileOnce(t *testing.T) {
+	processors := map[string]string{
+		"batch": `
+  batch:
+`,
+	}
+	resultDir, err := filepath.Abs(path.Join("results", t.Name()))
+	require.NoError(t, err)
+	sender := datasenders.NewFileLogWriter()
+	receiver := testbed.NewOTLPDataReceiver(testutil.GetAvailablePort(t))
+	loadOptions := testbed.LoadOptions{
+		DataItemsPerSecond: 1,
+		ItemsPerBatch:      10000000,
+		Parallel:           1,
+	}
+
+	// Write data at once, before starting up the collector
+	dataProvider := testbed.NewPerfTestDataProvider(loadOptions)
+	dataItemsGenerated := atomic.Uint64{}
+	dataProvider.SetLoadGeneratorCounters(&dataItemsGenerated)
+	ld, _ := dataProvider.GenerateLogs()
```

Comment:
is that returning an error? Can we check it?

Reply:
It actually returns a boolean, not an error.